addons/edi_sale/models/edi_sale_line_request_record.py

```diff
@@ -17,6 +17,15 @@ class EdiDocument(models.Model):
         "edi.sale.line.request.record", "doc_id", string="Sale Line Requests"
     )
 
+    @api.depends("sale_line_request_ids", "state")
+    def _compute_edi_rec_count(self):
+        if self.sale_line_request_ids:
+            for doc in self:
+                doc.total_records = len(doc.sale_line_request_ids)
```

Comment:
Wouldn't this conflict with the totals from other related fields on the EDI document? E.G. edi_sale_request_records.

Reply:
This one is tricky. There are some specialized classes with reference fields to both header & item records where we calculated the total of both; but for some classes like the `edi_sale` the header & item records are in different classes. It seems we can have only one of those; unless we find a clever way...

Reply:
Just to clarify: the sale lines method will get called first and since it doesn't propagate `super()` call, the sale request method wouldn't get called anyway. So we may leave it as is.